azavea/districtbuilder-2
# Update process.sh to take params and lookup state

## Overview
I was going to set up another prototype test site when I realized the `process.sh` script could be made more general with less manual effort to look up the state code and change the hard coded values where needed. Overall, this was probably unnecessary, but it allowed me to flex my scripting muscles again and learn about bash scripting. The changes comprise mainly of:
* Adding `state-lookup.conf` as a file holds all the necessary 2-digit state codes.
* Adding argument support process.sh. It now takes:
  * `-s`, the two-letter state abbreviation to look up
  * `-d`, a path to an input directory where the state geojson data is stored
* Modifying the output to a state-specific directory so it doesn't have to remove the previous output entirely
* Generalization of the output paths that are used in the mapshaper/node/tippecanoe etc. commands
* Looping through the repetitive label filtering
* Additional print statements indicating when things are happening

## Testing Instructions
* `cd public/process/`
* Copy contents of `data/output/` from [districtbuilder-data-pipeline](https://github.com/azavea/districtbuilder-data-pipeline) to `public/process/input/`
  * i.e. `geojson/bg/[state].geojson` and `geojson/c/[state].geojson`. You can just copy the whole contents of `/output/`
* Run `process.sh` with `-s` to specify any state you have data for
  * e.g. `bash process.sh -s NC`
* See output in `_output/[state]`. This contents of `/upload` can then be copied to the necessary location.
* Run it again for the same state. It will prompt to confirm overwriting
* Run it for a state that doesn't exist or is misspelled. It will complain and exit.
